packages/yoroi-extension/app/components/wallet/assets/NFTsList.js

```diff
@@ -109,26 +168,80 @@ function NfTsList({ list, intl }: Props & Intl): Node {
 
 export default (injectIntl(NfTsList): ComponentType<Props>);
 
-function NftCardImage({ image, name }) {
-  let nftImage
-  if (image != null) {
-    const ipfsHash = image.replace('ipfs://', '');
-    nftImage = (<img src={`https://ipfs.io/ipfs/${ipfsHash}`} alt={name} loading="lazy" />);
-  } else {
-    nftImage = (<DefaultNFT />);
-  }
+function isImageExist(imageSrc: string, onload: void => void, onerror: void => void) {
+  const img = new Image();
+  img.onload = onload;
+  img.onerror = onerror;
+  img.src = imageSrc;
+}
+
+export function NftImage({ imageUrl, name, width, height }: {|
+  imageUrl: string | null,
+  name: string,
+  width: string,
+  height: string
+|}): Node {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  imageUrl = imageUrl !== null ? imageUrl.replace('ipfs://', 'https://ipfs.io/ipfs/'): null;
+
+  useEffect(() => {
+    if (imageUrl !== null)
+      isImageExist(
+        String(imageUrl),
+        () => { setLoading(false); setError(false) }, // on-success
+        () => { setLoading(false); setError(true) }, // on-error
+      )
+  }, [imageUrl])
 
+  if (error || imageUrl === null) return (
+    <Box sx={{
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      backgroundColor: 'var(--yoroi-palette-gray-100)',
+      width,
+      height,
+    }}
+    >
+      <DefaultNFT />
+    </Box>
+  )
+
+  if (loading) return <Skeleton variant='rectangular' animation='wave' sx={{ width, height }} />
```

Comment:
Return the skeleton while loading before the error/imageUrl check.

Reply:
it is possible that the image NFT might have no URL due to missing metadata. that is why I am returning the `imageUrl` check first,